Frontend/HsTypeChecker.hs

```diff
@@ -990,6 +1016,13 @@ elabProgram theory (PgmData data_decl pgm) = do
   let fc_program = FcPgmDataDecl fc_data_decl fc_pgm
   return (fc_program, ty, final_theory)
 
+-- Elaborate a function declaration
```

Comment:
Note that we write function comments using `-- | `, in order to be haddock compatible